src/write_gcm.c

```diff
@@ -3568,15 +3568,20 @@ static void *wg_process_queue(wg_context_t *ctx, wg_queue_t *queue,
     }
     if (wg_update_stats(stats) != 0) {
       wg_some_error_occured_g = 1;
-      WARNING("%s: wg_update_stats failed.", this_plugin_name);
+      ERROR("%s: wg_update_stats failed.", this_plugin_name);
       break;
```

Comment:
Thanks, Corey. We're trying to distinguish catastrophic failures (not enough memory to run the consumer thread) from potentially transient failures (one point couldn't be processed for some reason). In this case, changing a `break` to a `continue` should actually keep your original readability properties, but prevent the consumer thread from exiting on anything non-catastrophic. We'd still log, but if we can keep the thread going, once the memory pressure eases off, the agent should recover without the user having to restart it.